src/librustc_mir/borrow_check/mod.rs

```diff
@@ -1110,6 +1112,35 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
                 );
                 return;
             }
+
+        }
+
+        if let &Place::Projection(_) = place {
```

Comment:
Comment:

---

FIXME(#21232). For now, error if assigning to a projection from an uninitialized local variable -- so e.g. doing `a.b = 22` when `a` is not yet initialized is simply an error. In the future, we could sometimes support this.